integration/library_test.go

```diff
@@ -0,0 +1,121 @@
+// +build integration
+
+package integration
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/extractor/docker"
+	"github.com/stretchr/testify/require"
+
+	_ "github.com/aquasecurity/fanal/analyzer/command/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/library/bundler"
+	_ "github.com/aquasecurity/fanal/analyzer/library/cargo"
+	_ "github.com/aquasecurity/fanal/analyzer/library/composer"
+	_ "github.com/aquasecurity/fanal/analyzer/library/npm"
+	_ "github.com/aquasecurity/fanal/analyzer/library/pipenv"
+	_ "github.com/aquasecurity/fanal/analyzer/library/poetry"
+	_ "github.com/aquasecurity/fanal/analyzer/library/yarn"
+	_ "github.com/aquasecurity/fanal/analyzer/os/alpine"
+	_ "github.com/aquasecurity/fanal/analyzer/os/amazonlinux"
+	_ "github.com/aquasecurity/fanal/analyzer/os/debianbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/photon"
+	_ "github.com/aquasecurity/fanal/analyzer/os/redhatbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/suse"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/dpkg"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpm"
+
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/types"
+)
+
+func TestFanal_Library_DockerMode(t *testing.T) {
+	ctx := context.Background()
+	d, _ := ioutil.TempDir("", "TestFanal_Library_*")
+	defer func() {
+		_ = os.RemoveAll(d)
+	}()
+	c, _ := cache.New(d)
+
+	opt := types.DockerOption{
+		Timeout:  600 * time.Second,
+		SkipPing: true,
+	}
+
+	ext, err := docker.NewDockerExtractor(opt, c)
+	require.NoError(t, err)
+	ac := analyzer.Config{Extractor: ext}
+
+	expectedFiles := []string{"etc/alpine-release", "etc/os-release", "lib/apk/db/installed", "/config"}
+	actualFiles, err := ac.Analyze(ctx, "alpine")
```

Comment:
From where is this `alpine`? If you expect this image is in Docker Engine, we should import the image into Docker Engine before running `ac.Analyze`.

Reply:
Yes, that's an oversight on my part. I'll fix it.

Reply:
As we discussed [before](https://github.com/aquasecurity/trivy/pull/335/files#diff-272fbc9a7f1f9b71135958c65d0091a6R61), we need to use a unique tag. If we don't do it, `fanal` automatically fetches an image from Docker Registry. I think this test in CI should fail, but it passed by accident. I'm not sure, but it might be because `alpine` is downloaded from Docker Hub.

Reply:
Yeah, I've taken care of this now.